src/core/trtserver.h

```diff
@@ -646,6 +646,33 @@ TRTSERVER_EXPORT TRTSERVER_Error* TRTSERVER_LoadModel(
 TRTSERVER_EXPORT TRTSERVER_Error* TRTSERVER_UnloadModel(
     TRTSERVER_Server* server, const char* model_name);
 
+/// Register a shared memory region on the inference server. If the shared
+/// memory region is already registered, it will be unregistered and then
+/// re-registered. Returned error indicates if shared memory region was
+/// registered successfully or not.
+/// \param server The inference server object.
+/// \param name The user-given name for the shared memory region to be
+/// registered.
+/// \param shm_key The unique name of the location in shared memory being
+/// registered.
+/// \param offset The offset into the shared memory region.
+/// \param byte_size The size, in bytes of the tensor data.
+/// \return a TRTSERVER_Error indicating success or failure.
+TRTSERVER_EXPORT TRTSERVER_Error* TRTSERVER_RegisterSharedMemory(
+    TRTSERVER_Server* server, const char* name, const char* shm_key,
+    const size_t offset, const size_t byte_size);
+
+/// Unregister a registered shared memory region on the inference server. If
+/// the shared memory region is not registered, do nothing and return failure.
```

Comment:
If not register don't return error, just silently ignore and return success. They can use status API if they want to know what is registered.